src/error.rs

```diff
@@ -162,3 +162,60 @@ impl From<untrusted::EndOfInput> for Error {
         Error::BadDer
     }
 }
+
+// Return the more specific of two errors.
+pub(crate) fn most_specific_error(prev: Error, new: Error) -> Error {
```

Comment:
For this too, I think it would make sense to make it `Error::most_specific()`.

Reply:
Done.

Reply:
Though re-reading your comment, I made this a method like `pub(crate) fn most_specific(self, new: Error) -> Error ...` instead of a static fn (_is that the right Rust term?_) like `pub(crate) fn most_specific(prev: Error, new: Error) -> Error ...`

Do you have a preference between the two?

Reply:
Yeah -- my preference (like yours) is with the associated method that takes a `self` param (and that was my intention).

Reply:
Great :-)